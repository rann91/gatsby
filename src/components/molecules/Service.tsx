/* @jsx jsx */
import { Fragment } from 'react'
import { AspectImage, Box, Heading, jsx, Text } from 'theme-ui'
import { rem } from '../../gatsby-plugin-theme-ui/index'
import { Strapi_ComponentSharedService } from '../../typings/graphql'

const Service = ({
  icon,
  title,
  description
}: Strapi_ComponentSharedService) => (
  <Fragment>
    {icon && (
      <Box
        mb={5}
        mx={['auto', null, null, 0]}
        sx={{ width: rem(50), height: rem(50) }}>
        <AspectImage
          ratio={1}
          src={icon?.file?.publicURL || undefined}
          alt={icon?.alternativeText || 'Service logo'}
          sx={{
            borderRadius: 4
          }}
        />
      </Box>
    )}
    {title && (
      <Heading as="h3" pb={3} sx={{ fontSize: 3 }}>
        {title}
      </Heading>
    )}
    {description && (
      <Text as="p" variant="p">
        {description}
      </Text>
    )}
  </Fragment>
)

export default Service
