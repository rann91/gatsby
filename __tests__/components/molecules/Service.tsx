import { render } from '@testing-library/react'
import React from 'react'
import Service from '../../../src/components/molecules/Service'
import service from '../../fixtures/service-single.json'

describe('Service', () => {
  it('renders correctly', () => {
    const { container } = render(<Service {...service} />)

    expect(container).toMatchSnapshot()
  })
})
