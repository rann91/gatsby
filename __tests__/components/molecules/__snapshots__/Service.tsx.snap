// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service renders correctly 1`] = `
<div>
  <div
    class="css-5uwokq-Service"
  >
    <div
      class="css-i9l3dq"
    >
      <div
        class="css-tmeqbf"
      />
      <div
        class="css-ojfau8"
      >
        <img
          alt="Service logo"
          class="css-1vz0lri-Service"
        />
      </div>
    </div>
  </div>
  <h3
    class="css-j6wujt-Service"
  >
    Management
  </h3>
  <p
    class="css-vurnku"
  >
    Blind texts it is an almost unorthographic life One day however a small line of blind text by the name.
  </p>
</div>
`;
